src/client/runner/automation/playback/move.js

```diff
@@ -160,27 +160,50 @@ export default function (to, inDragging, options, actionCallback, currentDocumen
 
     async.series({
         scrollToTarget: function (callback) {
-            var elementOffset = domUtils.isDomElement(to) ? positionUtils.getOffsetPosition(to) : null;
+            var isDomElement     = domUtils.isDomElement(to);
+            var scrollElement    = isDomElement ? to : document.documentElement;
+            var offsets          = null;
+            var elementRectangle = positionUtils.getElementRectangle(scrollElement);
+
+            if (isDomElement) {
+                offsets = {
+                    offsetX: typeof options.offsetX === 'number' ?
+                             options.offsetX : Math.round(elementRectangle.width / 2),
+                    offsetY: typeof options.offsetY === 'number' ?
+                             options.offsetY : Math.round(elementRectangle.height / 2)
+                };
+            }
+            else {
+                offsets = {
+                    offsetX: to.x,
+                    offsetY: to.y
+                };
+            }
 
-            scrollPlaybackAutomation(domUtils.isDomElement(to) ? to : targetPoint, options, currentDocument, function () {
-                if (domUtils.isDomElement(to)) {
-                    var newElementOffset = positionUtils.getOffsetPosition(to),
-                        elementScroll    = styleUtils.getElementScroll(to);
+            var scrollAutomation = new ScrollAutomation(scrollElement, offsets);
 
-                    if (to !== document.documentElement) {
-                        targetPoint.x += newElementOffset.left - elementOffset.left;
-                        targetPoint.y += newElementOffset.top - elementOffset.top;
-                    }
+            scrollAutomation
+                .run()
+                .then(() => {
+                    var elementOffset = isDomElement ? positionUtils.getOffsetPosition(to) : null;
 
-                    if (!/html/i.test(to.tagName) && styleUtils.hasScroll(to)) {
-                        targetPoint.x -= elementScroll.left;
-                        targetPoint.y -= elementScroll.top;
-                    }
-                }
-                targetScreenPoint = positionUtils.offsetToClientCoords(targetPoint);
+                    if (isDomElement) {
+                        var newElementOffset = positionUtils.getOffsetPosition(to),
+                            elementScroll    = styleUtils.getElementScroll(to);
+
+                        if (to !== document.documentElement) {
+                            targetPoint.x += newElementOffset.left - elementOffset.left;
+                            targetPoint.y += newElementOffset.top - elementOffset.top;
```

Comment:
Does this make sense? ```newElementOffset.left - elementOffset.left;```
Because earlier elementOffset  calculate before scrolling but now after like newElementOffset